core/src/main/scala/org/apache/spark/deploy/ApplicationDescription.scala

```diff
@@ -23,7 +23,9 @@ private[spark] class ApplicationDescription(
     val memoryPerSlave: Int,
     val command: Command,
     var appUiUrl: String,
-    val eventLogDir: Option[String] = None)
+    val sparkVersion: String,
```

Comment:
just adding an extra field seems like a solution that involves a smaller change, but let me know if you'd like me to change it.